mkdocs/docs/expression-dsl.md

```diff
@@ -151,20 +151,6 @@ age_in_range = Not(
 )
 ```
 
-### Type Safety
```

Comment:
I think this is confusing, since this is allowed:

![image](https://github.com/user-attachments/assets/2ea32248-4661-48f4-8c26-ee9085b32526)